File: LC/5.longestPalidrome.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    let l = 0,r = 0,max_length = 0,ans = [],n = s.length,len = 1,start = 0;
    for (let i = 0;i < n;i++) {
        l = i - 1,r = i + 1;
        while (l >= 0 && s[l] == s[i]) {
            len++;
            l--;
        }
        while (r < n && s[r] == s[i]) {
            len++;
            r++;
        }
        while (l >= 0 && r < n && s[r] == s[l]) {
            len += 2;
            l--;
            r++;
        }
        if (len > max_length) {
            max_length = len;
            start = l;
        }
        len = 1;
    }
    return s.substr(start + 1,max_length);
};

File: LC/5.longestPalidrome.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int n = s.size(),l = 0,r = 0,length = 1,index = 0,max_length = 0;
        for (int i = 0;i < n;i++) {
            l = i - 1;
            r = i + 1;
            while (l >= 0 && s[l] == s[i]) {
                length++;
                l--;
            }
            while (r < n && s[r] == s[i]) {
                length++;
                r++;
            }
            while (l >= 0 && r < n && s[l] == s[r]) {
                length += 2;
                l--;
                r++;
            }
            if (length > max_length) {
                max_length = length;
                index = l;
            }
            length = 1;
        }
        return s.substr(index + 1,max_length);
    }
};
